### PR TITLE
Add LocaleProvider

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 ## 1.1.1 under development
 
-- no changes in this release.
+- New #55: Add `LocaleProvider` (samdark)
+- Chg #55: Raise PHP version to 8 (samdark)
```
Comment (xepozz) [inline CHANGELOG.md:6]: Raise major version then

File: src/LocaleProvider.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+final class LocaleProvider
+{
+    private ?Locale $locale = null;
+
+    public function __construct(
+        private Locale $defaultLocale
+    ) {
+    }
+
+    public function get(): Locale
```
Comment (xepozz) [inline src/LocaleProvider.php:20]: Is there any use cases when a developer need to know was locale changed from default?

Comment (xepozz) [reply src/LocaleProvider.php:20]: Kind of `wasChanged(): bool => $this->locale !== null`

**Review by xepozz — CHANGES_REQUESTED** [review]

File: src/LocaleProvider.php (hunk 2 above)
Comment (samdark) [reply src/LocaleProvider.php:20]: Yes. For the purpose we have `getDefaultLocale()`.

**Review by xepozz — APPROVED** [review]

**Pull request merged**